physlr/physlr.py

```diff
@@ -1653,18 +1738,59 @@ def determine_molecules_process(u):
         """
         return Physlr.determine_molecules(Physlr.graph, u, Physlr.junctions, Physlr.args.strategy)
 
+    @staticmethod
+    def set_settings(round_num):
+        """
+        set settings for a pre-defined version of iterative molecule separation.
+        """
+        if round_num == 1:
+            print("Settings for round 1:", file=sys.stderr)
+
+            print(" # threads: ", Physlr.args.threads, "| sqcosbin t:", Physlr.args.sqcost,
+                  "\n |  junction_depth ", Physlr.args.junction_depth,
+                  " |  prune_junctions ", Physlr.args.prune_junctions,
+                  "\n | Skip small:", Physlr.args.skip_small,
+                  file=sys.stderr)
+        if round_num == 2:
+            print("Settings for round 2:", file=sys.stderr)
+
+            Physlr.args.threads = math.ceil(Physlr.args.threads / 8)
+            Physlr.args.junction_depth = 10
+            Physlr.args.cost = 0.55
```

Comment:
These are very specific settings for the different rounds -- how did we come to them and will they remain hard-coded?

Reply:
These settings are experimental and are only used for rounds 2+ of molecule separation.
`Physlr.args.threads =..` I have mentioned in the first comment of this PR. I can alternatively set to 1.
`Physlr.args.cost = 0.55` and `Physlr.args.sqcost` are experimental and we will automatically determine the threshold for cosine similarity in a later update.
`Physlr.args.junction_depth = 10`: seems to be a good value experimentally and has low dependency on input dataset.